syntax = "proto3";
 
package transport_catalogue_protobuf;
 
message Point {
    double x = 1;
    double y = 2;
}
 
message Rgb {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
}
 
message Rgba {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;
    double opacity = 4;
}
 
message Color {
    oneof colors {
        Rgb rgb = 1;
        Rgba rgba = 2;
        string string_color = 3;
        
        bool none = 4;
    }
}